plugins/cpu/fu-plugin-cpu.c

```diff
@@ -73,6 +76,17 @@ fu_plugin_add_security_attrs_intel_cet (FuPlugin *plugin, FuSecurityAttrs *attrs
 		return;
 	}
 
+	/* check that userspace has been compiled for CET support */
+	toolfn = g_build_filename (FWUPD_LIBEXECDIR, "fwupd", "fwupd-detect-cet", NULL);
+	if (!g_spawn_command_line_sync (toolfn, NULL, NULL, &exit_status, &error_local)) {
+		g_warning ("failed to test CET: %s", error_local->message);
+		return;
+	}
+	if (!g_spawn_check_exit_status (exit_status, &error_local)) {
+		g_debug ("CET does not function, not supported: %s", error_local->message);
+		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_UNUSED);
```

Comment:
Given how much it actually takes to make a CET measurement work (correct kernel patches, correct glibc, and fwupd supporting it) i'm not sure the unused flag makes any sense.

If I had the correct kernel support that the flags showed up (`ibt` and `shstk`) but hte missing glibc it wouldn't have booted.  If I had the correct glibc but fwupd was not compiled with support, then I wouldn't have this check in the first place.

Reply:
> i'm not sure the unused flag makes any sense

My logic for the extra data was to signify if the support was enabled in the OS; as I understand it you would still have cet in cpuinfo, but if glibc wasn't compiled with cet then it would essentially be unused. Is that wrong?

Reply:
I could only boot into single user mode and a ton of my binaries wouldn't execute without it being in glibc.

Reply:
So, forgive the stupid question, does that mean a CPU with CET enabled is unable to boot older distros like Ubuntu LTS?

Reply:
Well older distros that don't have CET enabled in the kernel won't export `ibt` or `shstk` in `/proc/cpuinfo`.  It was only after I booted the CET enabled kernel that those start to export.

Reply:
So maybe we don't need this PR at all? i.e. stick to what we were doing before?

Reply:
Interesting thought.  I really like a test that the shadow stack really is working though.

Reply:
Maybe we just fail the check with a runtime `!`?

Reply:
Majority of applications are compatible with shadow stack.   Here is an example of shadow stack:

https://gitlab.com/cet-software/cet-smoke-test/-/tree/master/violations

Reply:
>  I really like a test that the shadow stack really is working though

Yes, agreed. I think we either need the "unused" thing or a different runtime attr for CET. I'm happy to do either. The different attributes are already being suggested for DCI lock/enabled so maybe this isn't so bad.

Reply:
You don't think setting CET to invalid would be appropriate in this case?

Reply:
No, invalid would fail the HSI test which makes it look like the OEM is at fault, when really it's the users system that is screwed up or a mixture of old and new binaries.

Reply:
I've updated the branch with a much better check, splitting these out like we did the others. It's also better as now CET being available but not being used by the OS is a *runtime* failure, not something that affects the HSI value.